examples_tests/51.RadixSort/main.cpp

```diff
@@ -0,0 +1,287 @@
+		nullptr, nullptr, nullptr, nullptr);
+
+	RadixSortClass::Parameters_t sort_push_constants[RadixSortClass::PASS_COUNT];
+	RadixSortClass::DispatchInfo_t sort_dispatch_info;
+
+	const uint32_t upsweep_pass_count = (total_scan_pass_count / 2) + 1;
+	core::vector<ScanClass::Parameters_t> scan_push_constants(upsweep_pass_count);
+	core::vector<ScanClass::DispatchInfo_t> scan_dispatch_info(upsweep_pass_count);
+
+	RadixSortClass::buildParameters(in_gpu_range.size / sizeof(SortElement), WG_SIZE, sort_push_constants, &sort_dispatch_info,
+		scan_push_constants.data(), scan_dispatch_info.data());
+
+	SBufferRange<IGPUBuffer> scratch_gpu_range = { 0 };
+	scratch_gpu_range.size = in_gpu_range.size;
+	scratch_gpu_range.buffer = driver->createDeviceLocalGPUBufferOnDedMem(in_gpu_range.size);
+
+	const uint32_t histogram_count = sort_dispatch_info.wg_count[0] * RadixSortClass::BUCKETS_COUNT;
+	SBufferRange<IGPUBuffer> histogram_gpu_range = { 0 };
+	histogram_gpu_range.size = histogram_count * sizeof(uint32_t);
+	histogram_gpu_range.buffer = driver->createDeviceLocalGPUBufferOnDedMem(histogram_gpu_range.size);
+
+	ScanClass::updateDescriptorSet(ds_scan, histogram_gpu_range, driver);
+
+	for (uint32_t pass = 0; pass < RadixSortClass::PASS_COUNT; ++pass)
+	{
+		const uint32_t count = 2u;
+		asset::SBufferRange<IGPUBuffer> descriptor_ranges[count];
+		if (pass % 2)
+		{
+			descriptor_ranges[0] = scratch_gpu_range;
+			descriptor_ranges[1] = in_gpu_range;
+		}
+		else
+		{
+			descriptor_ranges[0] = in_gpu_range;
+			descriptor_ranges[1] = scratch_gpu_range;
+		}
+
+		RadixSortClass::updateDescriptorSet(ds_histogram, descriptor_ranges, count, driver);
+		{
+			const IGPUDescriptorSet* descriptor_sets[2] = { ds_scan, ds_histogram };
+			driver->bindDescriptorSets(video::EPBP_COMPUTE, upsweep_pipeline->getLayout(), 0u, 2u, descriptor_sets, nullptr);
+		}
+
+		driver->bindComputePipeline(histogram_pipeline);
+		RadixSortClass::dispatchHelper(histogram_pipeline->getLayout(), sort_push_constants[pass], sort_dispatch_info, driver);
+
+		for (uint32_t scan_pass = 0; scan_pass < upsweep_pass_count; ++scan_pass)
+		{
+			driver->bindComputePipeline(upsweep_pipeline);
+			ScanClass::dispatchHelper(upsweep_pipeline->getLayout(), scan_push_constants[scan_pass], scan_dispatch_info[scan_pass], driver);
+		}
+
+		for (uint32_t scan_pass = upsweep_pass_count; scan_pass < total_scan_pass_count; ++scan_pass)
+		{
+			driver->bindComputePipeline(downsweep_pipeline);
+			ScanClass::dispatchHelper(downsweep_pipeline->getLayout(), scan_push_constants[total_scan_pass_count - 1 - scan_pass], scan_dispatch_info[total_scan_pass_count - 1 - scan_pass], driver);
+		}
+
+		RadixSortClass::updateDescriptorSet(ds_scatter, descriptor_ranges, count, driver);
+		{
+			const IGPUDescriptorSet* descriptor_sets[2] = { ds_scan, ds_scatter };
+			driver->bindDescriptorSets(video::EPBP_COMPUTE, upsweep_pipeline->getLayout(), 0u, 2u, descriptor_sets, nullptr);
+		}
```

Comment:
Here I wanted to reuse `ds_histogram` since both of them the exact same `descriptor_ranges`, but couldn't because, for some reason the input and output buffers weren't being bound in ping-pong, odd-even fashion ie input buffer for pass 1 remained input buffer of pass 0 instead of being output buffer of pass 0 and so on

Reply:
you cannot have a `updateDescriptorSet` in the middle of a compute loop

the only thing in a execution function/loop that should be there are:
- bind pipeline
- bind descriptor set
- push constants
- dispatch/draw
- barrier

Reply:
create two copies of each descriptor set for pingponging

Reply:
> but couldn't because, for some reason the input and output buffers weren't being bound in ping-pong
if you update a descriptor set while its bound dont expect the binding to change

In fact that will 100% crash under Vulkan.